### PR TITLE
Code Homogenisation and Bugfixes

### PR DESCRIPTION
**Pull request by ARamsay17**
Code homogenisation started on PIU. Factory and Default methods combined into Class_Based and Class_Based_L2 to avoid the repetition of code. Band convolution for FRM and Class-Based regimes converted into two callable methods (one for each regime) which are called sequentially for each sensor, only convolving to sensor bands indicated in the config. 

Comments added to PIU in order to explain some non-intuitive features (this effort will continue).

Bugfixes for issues #234, #254 and a partial solution to #253

### CODE REVIEW

File: Source/ProcessL1aqc.py
Comment (oceancolorcoder) [inline Source/ProcessL1aqc.py:1]: Hi @ARamsay17, why was this changed back again?

Comment (oceancolorcoder) [reply Source/ProcessL1aqc.py:1]: I assume this was done by accident again and will be reverting it to the 'DATETIME' tests. Please let me know if there is a reason to go back to the old 'SOLARTRACKER_STATUS' test.

Comment (ARamsay17) [reply Source/ProcessL1aqc.py:1]: @oceancolorcoder, I thought the merge with upstream dev would have solved this issue, or at least flagged it. Yes your intuition is correct I have no reason to go back to the old solution.

Comment (ARamsay17) [reply Source/ProcessL1aqc.py:1]: Thanks @oceancolorcoder, I will include this information in the PR going forwards. I have tested on all 3 regimes for the PySAS and TriOS sample data, plotting the uncertainties and comparing to previous versions of the code. In addition this PR was tested on Krista's and Raphael's data that's available on the box.  All testing was done on Linux as my windows PC is not able to meet the demands of running HyperCP anymore. I am working on a solution to this.

Comment (oceancolorcoder) [reply Source/ProcessL1aqc.py:1]: Excellent! Thank you @ARamsay17
